### PR TITLE
[NETBEANS-3581] Improvements to DropDownButton/DropDownToggleButton

### PR DESCRIPTION
**Pull request by eirikbakke**
DropDownButton/DropDownToggleButton are most commonly used in the various NetBeans toolbars--there is one pair of DropDownButton buttons in the editor toolbar, for instance.

(I use both DropDownButton and DropDownToggleButton in my NetBeans Platform application, and have tested both there. I have tested DropDownButton in the IDE as well. There are no current uses of DropDownToggleButton in the IDE.)

Details:
* Copy an old bugfix from DropDownButton to DropDownToggleButton (an added
  updateRollover call), to fix a problem where the latter would show the wrong
  icon after an icon change.
* Fix a bug where DropDownButton would occasionally get in a state where two
  mouse clicks were required to open it.
* Prevent dropdown button menus from showing up when the button is disabled.
* If DropDownButton has no action listeners registered, allow the user to click
  anywhere in the button area (not just the arrow area on the side) to open the
  dropdown menu.
* Review all differences between DropDownButton and DropDownToggleButton, and
  make sure they are legitimate. (These two classes are nearly identical, save
  for a few changes.)

dropdown button examples

### CODE REVIEW

File: platform/openide.awt/src/org/openide/awt/DropDownButton.java [hunk 1]
```diff
@@ -78,6 +79,7 @@ public DropDownButton( Icon icon, JPopupMenu popup ) {
         resetIcons();
 
         addPropertyChangeListener(  DropDownButtonFactory.PROP_DROP_DOWN_MENU,new PropertyChangeListener() {
```
Comment (lkishalmi) [inline platform/openide.awt/src/org/openide/awt/DropDownButton.java]: Lambda?

File: platform/openide.awt/src/org/openide/awt/DropDownButton.java [hunk 2]
```diff
@@ -152,22 +159,33 @@ public void mouseExited( MouseEvent e ) {
...
                     JPopupMenu menu = getPopupMenu();
                     if( null != menu ) {
                         menu.removePopupMenuListener( this );
                     }
+                    /* If the popup was closed by a mouse click inside the button area, the button
+                    may also receive a mousePressed event, although this seems not to be guaranteed.
+                    Ignore any such button press while the popup is closing, to avoid interpreting
+                    the press as a click to open the menu again. */
+                    popupClosingInProgress = true;
+                    SwingUtilities.invokeLater(new Runnable() {
```
Comment (lkishalmi) [inline platform/openide.awt/src/org/openide/awt/DropDownButton.java]: Lambda?

Comment (eirikbakke) [reply platform/openide.awt/src/org/openide/awt/DropDownButton.java]: Ah, sure. Didn't realize they existed in Java 8.

**Comment by eirikbakke** [comment]
I'm going to test this on MacOS too, before merging it.

**Comment by eirikbakke** [comment]
Latest version now tested on both Windows and MacOS. Merging.

**Pull request merged**